src/frontend/apps/impress/src/features/docs/doc-export/components/ModalExport.tsx

```diff
@@ -142,13 +144,122 @@ export const ModalExport = ({ onClose, doc }: ModalExportProps) => {
       });
 
       blobExport = await exporter.toODTDocument(exportDocument);
+    } else if (format === DocDownloadFormat.HTML) {
+      const editorHtml = await editor.blocksToHTMLLossy();
+
+      // Parse HTML and fetch media so that we can package a fully offline HTML document in a ZIP.
+      const domParser = new DOMParser();
+      const parsedDocument = domParser.parseFromString(editorHtml, 'text/html');
+
+      const mediaFiles: { filename: string; blob: Blob }[] = [];
+      const mediaElements = Array.from(
+        parsedDocument.querySelectorAll<
+          | HTMLImageElement
+          | HTMLVideoElement
+          | HTMLAudioElement
+          | HTMLSourceElement
+        >('img, video, audio, source'),
+      );
+
+      await Promise.all(
+        mediaElements.map(async (element, index) => {
+          const src = element.getAttribute('src');
+
+          if (!src) {
+            return;
+          }
+
+          const fetched = await exportCorsResolveFileUrl(doc.id, src);
+
+          if (!(fetched instanceof Blob)) {
+            return;
+          }
+
+          // Derive a readable filename:
+          // - data: URLs → use a generic "media-N.ext"
+          // - normal URLs → keep the last path segment as base name.
+          let baseName = `media-${index + 1}`;
+
+          if (!src.startsWith('data:')) {
+            try {
+              const url = new URL(src, window.location.origin);
+              const lastSegment = url.pathname.split('/').pop();
+              if (lastSegment) {
+                baseName = `${index + 1}-${lastSegment}`;
+              }
+            } catch {
+              // Ignore invalid URLs, keep default baseName.
+            }
+          }
+
+          let filename = baseName;
+
+          // Ensure the filename has an extension consistent with the blob MIME type.
+          const mimeType = fetched.type;
+          if (mimeType && !baseName.includes('.')) {
+            const subtype = mimeType.split('/')[1] || '';
+            let extension = '';
+
+            if (subtype.includes('svg')) {
+              extension = 'svg';
+            } else if (subtype.includes('jpeg') || subtype.includes('pjpeg')) {
+              extension = 'jpg';
+            } else if (subtype.includes('png')) {
+              extension = 'png';
+            } else if (subtype.includes('gif')) {
+              extension = 'gif';
+            } else if (subtype.includes('webp')) {
+              extension = 'webp';
+            } else if (subtype.includes('pdf')) {
+              extension = 'pdf';
+            } else if (subtype) {
+              extension = subtype.split('+')[0];
+            }
+
+            if (extension) {
+              filename = `${baseName}.${extension}`;
+            }
+          }
+
+          element.setAttribute('src', filename);
+          mediaFiles.push({ filename, blob: fetched });
+        }),
+      );
+
+      const lang = i18next.language || 'fr';
```

Comment:
```suggestion
      const lang = i18next.language || fallbackLng;
```

From: https://github.com/suitenumerique/docs/blob/999c81499a087214c8de72575f202bbfc5e015a1/src/frontend/apps/impress/src/i18n/config.ts#L1-L2